commands/src/main/java/pl/fratik/commands/MemberListener.java

```diff
@@ -115,6 +112,46 @@ public void onMessage(MessageReceivedEvent m) {
             // nie mamy permów, zawijamy się
         }
     }
+
+    @Subscribe
+    public void onRoleAdd(GuildMemberRoleAddEvent e) {
+        updateNickname(e.getMember());
+    }
+    @Subscribe
+    public void onRoleRemmove(GuildMemberRoleRemoveEvent e) {
+        updateNickname(e.getMember());
+    }
+
+    private void updateNickname(Member mem) {
+        GuildConfig gc = getGuildConfig(mem.getGuild());
+        if (gc.getRolePrefix() == null) return;
+        for (Role role : mem.getRoles()) {
+            String prefix = gc.getRolePrefix().get(role.getId());
+            if (prefix == null) continue;
+            
+            AtomicReference<String> nick = new AtomicReference<>(mem.getNickname());
+            if (nick.get() == null) nick.set(mem.getUser().getName());
+            else {
+                gc.getRolePrefix().values().forEach(p -> {
```

Comment:
zastąpiłbym `for (... : gc.getRolePrefix().values())`, nie trzeba by było kombinować z AtomicReference